sources/packages/web/src/services/InstitutionUserService.ts

```diff
@@ -0,0 +1,250 @@
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        .map((authorization) => authorization.location!.name);
+      locations.push(...locationNames);
+    }
+    return {
+      institutionUserId: institutionUser.id,
+      email: institutionUser.user.email,
+      userName: institutionUser.user.userName,
+      displayName: institutionUser.user.userFullName,
+      locations,
+      userType,
+      roles,
+      isActive: institutionUser.user.isActive,
+      disableRemove:
+        AuthService.shared.userToken?.userName ===
+        institutionUser.user.userName,
+    };
+  }
+
+  /**
+   * Get institution user summary.
+   * @param paginationOptions pagination options.
+   * @param institutionId institution to be searched. If not
+   * provided the institution id will be retrieved from the token,
+   * when available.
+   * @returns all filtered institution users.
+   */
+  async institutionUserSummary(
+    paginationOptions: PaginationOptions,
+    institutionId?: number,
+  ): Promise<InstitutionUserSummary> {
+    const response = await ApiClient.InstitutionUserApi.searchUsers(
+      paginationOptions,
+      institutionId,
+    );
+    return {
+      results: response.results.map((result) =>
+        this.convertToInstitutionUserViewModel(result),
+      ),
+      count: response.count,
+    };
+  }
+
+  /**
+   * Create the user authorizations to be associate with the user.
+   * @param isAdmin must be created as an admin user.
+   * @param isLegalSigningAuthority for admin users, define if the role
+   * legal-signing-authority should be associated with the user.
+   * @param locationAuthorizations for non-admin users, the individual permission
+   * for every location.
+   */
+  private createUserPermissions(
+    isAdmin: boolean,
+    isLegalSigningAuthority: boolean,
+    locationAuthorizations: LocationAuthorization[],
+  ): UserPermissionAPIInDTO[] {
+    let permissions: UserPermissionAPIInDTO[];
+    if (isAdmin) {
+      // User is an admin and will have access for all the locations.
+      permissions = [
+        {
+          userType: InstitutionUserTypes.admin,
+          userRole: isLegalSigningAuthority
+            ? InstitutionUserRoles.legalSigningAuthority
+            : undefined,
+        } as UserPermissionAPIInDTO,
+      ];
+    } else {
+      // User is not an admin and will have the permission assigned to the individual locations.
+      // Filter locations with access. At this point the UI validations already ensured
+      // that there will be at least one location defined with some access level.
+      permissions = locationAuthorizations
+        .filter(
+          (locationAccess) =>
+            locationAccess.userAccess === LocationUserAccess.User,
+        )
+        .map(
+          (locationAccess) =>
+            ({
+              locationId: locationAccess.id,
+              userType: locationAccess.userAccess,
+            } as UserPermissionAPIInDTO),
```

Comment:
not part of this PR, but we can remove `as UserPermissionAPIInDTO`

Reply:
It can be removed, but in the cases outside the controller DTOs, like this one, IMO it makes it easier to read.
Open for thoughts. 😉